### PR TITLE
feat(Header & Footer): Add header & footer to the dropdown list.

### PR DESCRIPTION
**Pull request by felixmosh**
This feature allows to add header & footer to the dropdowns for each of the default templates.
bootstrap
select2
selectize

### CODE REVIEW

**Comment by Jefiozie** [comment]
HI @felixmosh  
Thanks for submitting this PR, looks like it will help resolve #1323 but before I can review this issue could you clean up the PR, you committed the dist directory and that is not correct. Also I believe a rebase is needed for the last changes. 

Thanks  looking forward on the review.

**Comment by felixmosh** [comment]
Done.

File: src/common.css [hunk 1]
```diff
@@ -13,11 +13,10 @@
   overflow: hidden !important;
   position: absolute !important;
   outline: 0 !important;
-  left: 0px !important;
-  top: 0px !important;
+  left: 0 !important;
+  top: 0 !important;
```
Comment (Jefiozie) [inline src/common.css]: Any specific reason why you changed the 0px to 0?

Comment (felixmosh) [inline src/common.css]: Convention wise, in css if you have 0px it should be 0.

File: src/common.css [hunk 2]
```diff
@@ -26,68 +25,84 @@
...
-    border-color: #D44950;
+  border-color: #D44950;
 }
 
 .select2-result-single {
   padding-left: 0;
 }
 
-.select2-locked > .select2-search-choice-close{
-  display:none;
```
Comment (Jefiozie) [inline src/common.css]: Could you fix all the spacing in this file?

File: src/selectize/choices.tpl.html [hunk 3]
```diff
@@ -1,5 +1,4 @@
-<div ng-show="$select.open" class="ui-select-choices ui-select-dropdown selectize-dropdown"
```
Comment (Jefiozie) [inline src/selectize/choices.tpl.html]: Could you explain why you removed all the classes from this div?

File: test/select.spec.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-describe('ui-select tests', function() {
```
Comment (Jefiozie) [inline test/select.spec.js]: Could you fix the spacing int his file?

File: src/common.css (hunk 1 above)
Comment (Jefiozie) [inline src/common.css]: Is the important needed?

Comment (felixmosh) [inline src/common.css]: No

File: src/common.css (hunk 2 above)
Comment (felixmosh) [inline src/common.css]: I did that...

**Comment by Jefiozie** [comment]
Second could you have a look at the [wiki](https://github.com/angular-ui/ui-select/wiki/ui-select-header-and-footer) this is from a PR before. Maybe you can update it when this PR is merged.

File: src/common.css (hunk 2 above)
Comment (felixmosh) [inline src/common.css]: I did that, because of that you see some places that fixes the spacing.

File: src/selectize/choices.tpl.html (hunk 3 above)
Comment (felixmosh) [inline src/selectize/choices.tpl.html]: I`ve moved the dropdown classes and functionality to here https://github.com/angular-ui/ui-select/pull/1930/files/fdae8530b84f3426b3b411da2cf840248a09155a#diff-55a8341ed55143ae563b5f1e54237ca8R13

Comment (Jefiozie) [inline src/selectize/choices.tpl.html]: missed that one.

File: test/select.spec.js (hunk 4 above)
Comment (felixmosh) [inline test/select.spec.js]: Done

File: src/common.css (hunk 2 above)
Comment (felixmosh) [inline src/common.css]: Done

**Comment by felixmosh** [comment]
The wiki is up to date to the current implementation :).
I`ve added additional example how to use it in each of the templates.

**Comment by ahules** [comment]
How we can use header/footer now? I mean what URL we must add to bower.json?

**Comment by Jefiozie** [comment]
@ahules  Do you mean when this will be release so that you can use it with bower?

**Comment by ahules** [comment]
@Jefiozie yes, now I use TechNaturally/ui-select#feat-header-footer-bower. But that realisation of header/footer triggers click twice.
Maybe you can create some branch named "experimental" or smth. :)

**Comment by Jefiozie** [comment]
This will be released in the newer version of this repo. For now i'm not making any experimental branches.  As i need to maintain that branch also and it is hard enough to maintain the repo on it own.
If need you can use the code as committed here. 

Hope you understand.

**Comment by ahules** [comment]
@Jefiozie, Can you help? I will create fork and merge this commit manually?

**Comment by Jefiozie** [comment]
I don't have the time to help you having a hard time to maintaining this repo. 
I hope to update the lib by the end of this month so then you can use the official library for the functionality.

**Comment by Jefiozie** [comment]
Hi @felixmosh Could you resolve the conflict? (don't know why this is suddenly here) afte this i will merge the PR. Thanks for your help.

**Comment by felixmosh** [comment]
Done

**Comment by cyrilchapon** [comment]
This breaks 100% width and ul bullet inside ui-select-choices on my application.

See 

before (*0.19.6*)
image


after (*0.19.7*)
image

**Comment by felixmosh** [comment]
@Jefiozie why this PR was removed?

**Comment by felixmosh** [comment]
@Jefiozie, I've fixed that already, I will create a new PR.

**Comment by felixmosh** [comment]
Fresh PR https://github.com/angular-ui/ui-select/pull/1982

**Pull request merged**